shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -1024,6 +1086,18 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, val ne
       case pat if ctx.inPattern =>
         err(msg"Unsupported pattern shape${
           if (dbg) " ("+pat.getClass.toString+")" else ""}:", pat.toLoc)(raise)
+      case Lam(pat, body) if ctx.inQuote =>
+        println(s"TYPING QUOTED LAM")
+        ctx.enterQuote.poly { newCtx =>
```

Comment:
I still don't see why you want to reset `qenv` in this case. I think you're still misunderstanding why this does not do the right thing.

Do you really want the skolems for the two occurrences of the same `x` to be different, int he following example?

```
f => x `=> f(x, y `=> x)
//│ forall 'a 'b 'c. ((Var['a, in ??x out ??x0], Code[anything -> 'a, ??x0 & ~??y]) -> Code['b, 'c]) -> Code['a -> 'b, 'c]
//│ res
//│     = [Function: res]
```

You can see two distinct skolems were created. Why do you want that??